pallets/inflation/src/lib.rs

```diff
@@ -322,28 +346,100 @@ pub mod pallet {
+        ///
+        /// It might seem similar to forcing the inflation recalculation, but it's not.
+        /// This function adjusts the existing configuration, respecting the `max_emission` value used to calculate the current inflation config.
+        /// (The 'force' approach uses the current total issuance)
+        ///
+        /// This call should be used in case inflation parameters have changed during the cycle, and the configuration should be adjusted now.
+        pub(crate) fn readjusted_config() -> InflationConfiguration {
+            // 1. First calculate the params needed to derive the `max_emission` value used to calculate the current inflation config.
+            let config = ActiveInflationConfig::<T>::get();
+
+            // This is reuse from `new_config` function.
+            let blocks_per_cycle = Balance::from(T::CycleConfiguration::blocks_per_cycle().max(1));
+            let build_and_earn_eras_per_cycle =
+                Balance::from(T::CycleConfiguration::build_and_earn_eras_per_cycle().max(1));
+            let periods_per_cycle =
+                Balance::from(T::CycleConfiguration::periods_per_cycle().max(1));
+
+            // 2. Calculate reward pool amounts per cycle from the existing inflation configuration.
+            let collator_reward_pool = config
+                .collator_reward_per_block
+                .saturating_mul(blocks_per_cycle);
+
+            let treasury_reward_pool = config
+                .treasury_reward_per_block
+                .saturating_mul(blocks_per_cycle);
+
+            let dapp_reward_pool = config
+                .dapp_reward_pool_per_era
+                .saturating_mul(build_and_earn_eras_per_cycle);
+
+            let base_staker_reward_pool = config
+                .base_staker_reward_pool_per_era
+                .saturating_mul(build_and_earn_eras_per_cycle);
+            let adjustable_staker_reward_pool = config
+                .adjustable_staker_reward_pool_per_era
+                .saturating_mul(build_and_earn_eras_per_cycle);
+
+            let bonus_reward_pool = config
+                .bonus_reward_pool_per_period
+                .saturating_mul(periods_per_cycle);
+
+            // 3. Sum up all values to get the old `max_emission` value.
+            let max_emission = collator_reward_pool
+                .saturating_add(treasury_reward_pool)
+                .saturating_add(dapp_reward_pool)
+                .saturating_add(base_staker_reward_pool)
+                .saturating_add(adjustable_staker_reward_pool)
+                .saturating_add(bonus_reward_pool);
+
+            // 4. Calculate new inflation configuration
+            Self::new_config(config.recalculation_era, max_emission)
+        }
+
+        // Calculate new inflation configuration, based on the provided `max_emission`.
+        fn new_config(
+            recalculation_era: EraNumber,
+            max_emission: Balance,
```

Comment:
Design-wise, if we would inject InflationParams as the third parameter of this function, it could be slightly better.

Reply:
Same for `CycleConfiguration`, but I would understand that the direct access to those parameters are common place and the consistent style.

Reply:
`InflationParams` are available directly in the function, as well as `CycleConfiguration` so I don't see the need for that.